tdrs-backend/tdpservice/parsers/test/test_parse.py

```diff
@@ -1146,6 +1146,26 @@ def test_parse_ssp_section4_file(ssp_section4_file, dfs):
     assert first.RPT_MONTH_YEAR == 201810
     assert first.FAMILIES_MONTH == 748
 
+@pytest.fixture
+def ssp_section2_rec_oadsi_file(stt_user, stt):
+    """Fixture for sp_section2_rec_oadsi_file."""
+    return util.create_test_datafile('ssp_section2_rec_oadsi_file.txt', stt_user, stt, 'SSP Closed Case Data')
+
+
+@pytest.mark.django_db()
+def test_parse_ssp_section2_rec_oadsi_file(ssp_section2_rec_oadsi_file, dfs):
+    """Test parsing SSP Section 2 REC_OADSI."""
+    ssp_section2_rec_oadsi_file.year = 2019
+    ssp_section2_rec_oadsi_file.quarter = 'Q1'
+
+    dfs.datafile = ssp_section2_rec_oadsi_file
+
+    parse.parse_datafile(ssp_section2_rec_oadsi_file)
```

Comment:
you didn't do anything wrong, i'd just like to make one point about these tests being more integration-tests than unit tests.

here, we create a whole new data file and send it through the entire parsing routine just to check that a single field validator works. it'd be nice to have a test that only checks that `required=False` does what it's supposed to, then we can be confident in uses of `required=False` going forward without these large/complex integration tests (and not have to write a new test every time a arbitrary field's `required` value changes). smaller unit tests are also more resilient against side effects or changes in other parts of the system.

this is the primary intent behind #2641

Reply:
If we leave the test, I think you should also query appropriate record and verify that the field is `None`.